sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -178,11 +178,15 @@ object StateStoreMetrics {
 trait StateStoreCustomMetric {
   def name: String
   def desc: String
+  def unit: String
 }
 
-case class StateStoreCustomSumMetric(name: String, desc: String) extends StateStoreCustomMetric
-case class StateStoreCustomSizeMetric(name: String, desc: String) extends StateStoreCustomMetric
-case class StateStoreCustomTimingMetric(name: String, desc: String) extends StateStoreCustomMetric
+case class StateStoreCustomSumMetric(name: String, desc: String, unit: String)
```

Comment:
Sorry for the comment at the last minute. I think the PR is ready for merge. Only one concern for the API change.
This is an API change and will break extended StateStore if they have custom metrics. Seems now the unit is the UI only requirement, how about we generate the unit in `StreamingQueryStatisticsPage.scala`? Or we can add the unit info in the `desc` string. e.g "StateStoreCustomSizeMetric("uncompressedFileSize", "uncompressed file size - file bytes")" WDYT?

Reply:
How about set empty string as the default value for `unit` ? This can resolve the compatibility issue.

Reply:
Yeah agree, empty string can also resolve the issue.

Reply:
Nice finding! Glad I wait for 2 days to figure out any missing like this.

Btw, I have a deeper question. Why the state store (+provider) interfaces are in sql/execution which is considered as "private"? We know there're several 3rd party implementations, and placing these APIs in sql/execution package means that we don't guarantee the compatibility as public API. e.g. MiMa didn't complain this.

What would be the way to go? Could we move the interface with aliasing the old one, and deprecate the old one? Otherwise, should we make a breaking change to get better result on future?

Reply:
> Btw, I have a deeper question. Why the state store (+provider) interfaces are in sql/execution which is considered as "private"? We know there're several 3rd party implementations, and placing these APIs in sql/execution package means that we don't guarantee the compatibility as public API. e.g. MiMa didn't complain this.

I wondered why MiMa didn't find the compatibility issue too. As @HeartSaVioR says `sql/execution` seems to be considered as "private", but I guess it may not be considered that the package is private at that time those interfaces were introduced to the package.
Anyway, it seems that we need to discuss the guarantee of the compatibility in another place.

Reply:
More eyes, better code. I've just added default value as empty string.

Reply:
Ping me here if somebody doesn't agree on the default value, closing it for now.